src/Sylius/Bundle/PayumBundle/Validator/GatewayConfigurationValidator.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\PayumBundle\Validator;
+
+use Sylius\Bundle\PayumBundle\Validator\Constraints\GatewayConfig;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+final class GatewayConfigurationValidator extends ConstraintValidator
+{
+    /** @var array<string, mixed> */
+    private array $factoriesValidators;
+
+    /**
+     * @param array<string, string> $factoryNames
+     * @param array<string, mixed> $factoriesValidators
+     */
+    public function __construct(private array $factoryNames, iterable $factoriesValidators)
+    {
+        $this->factoriesValidators = $factoriesValidators instanceof \Traversable ? iterator_to_array($factoriesValidators) : $factoriesValidators;
+    }
+
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        /** @var GatewayConfig $constraint */
+        Assert::isInstanceOf($constraint, GatewayConfig::class);
+
+        if (!in_array($value->getFactoryName(), array_keys($this->factoryNames), true) || $value->getFactoryName() === 'offline') {
+            return;
+        }
+
+        $configuration = $value->getConfig();
+
+        /** @var GatewayConfigValidatorInterface $validator */
+        $validator = $this->factoriesValidators[$value->getFactoryName()];
```

Comment:
I am thinking about the proposal I made [here](https://github.com/Sylius/Sylius/pull/15551/files#diff-580a5b1905e1435e1a428a50bef01dd3c4374c2d9ae6a397c3263aace75367aeR39). We should run the validation here with the group(`factoryName`) and then include this in the xml.
```
<constraint name="Collection">
    <option name="groups">stripe_checkout</option>
    <option name="fields">
        <value key="secret_key">
            <constraint name="NotBlank">
                <option name="groups">stripe_checkout</option>
            </constraint>
        </value>
        <value key="publishable_key">
            <constraint name="NotBlank">
                <option name="groups">stripe_checkout</option>
            </constraint>
        </value>
    </option>
</constraint>
```
WDYT?